src/services/mongoose/schema.types.d.ts

```diff
@@ -163,3 +163,18 @@ type PaymentFlowStateRecordIndex = XOR<
   senderWalletId: string
   inputAmount: number
 }
+
+type WalletOnChainPendingReceiveRecord = {
+  walletId: string
+  address: string
+  transactionHash: string
+  amount: number
```

Comment:
If these are wallet amount & fee I might rename to `walletAmount`, `walletFee` and `walletCurrency` just to be clear that they're not sats denominated.

Also, should we persist satsAmount and satsFee explicitly too?

Reply:
Updated https://github.com/GaloyMoney/galoy/pull/2622/commits/903ca44ded470fca59c481cbf8fcebccb121948a